experimental/filesystem-backends.mdx

```diff
@@ -0,0 +1,332 @@
+
+  </Tab>
+</Tabs>
+
+## Backend Types
+
+There are currently two backend types we classify as being filesystem backends, `local` and `git`.
+The `local` backend has been primarily developed to support a local development experience.
+Whereas, the `git` backend is intended for production and serves flag state directly from a configurable repository and Git reference (e.g. branch).
+
+### `local` type
+
+The purpose of this backend type is to support serving Flipt flag state directly from your local filesystem.
+You can simply specify a relative or absolute directory in order to start a local Flipt instance and serve flag state.
+This is particularly useful for local development and validation of flag state changes.
+
+Flipt will periodically rebuild its state from the local disk.
+
+<Tabs>
+  <Tab title="Environment Variable">
+
+    ```bash
+    FLIPT_EXPERIMENTAL_FILESYSTEM_STORAGE_ENABLED=true
+    FLIPT_STORAGE_TYPE="local"
+    FLIPT_STORAGE_LOCAL_PATH="."
+    ```
+
+  </Tab>
+  <Tab title="Configuration Yaml">
+
+    ```yaml
+    experimental:
+      filesystem_storage:
+        enabled: true
+
+    storage:
+      type: local
+      local:
+        path: "."
+    ```
+
+  </Tab>
+</Tabs>
+
+> Serve your current directory.
+
+### `git` type
+
+The `git` type backend is used to configure a target git repository and git reference to source feature flag state from.
+The configuration contains fields for addressing the repository, configuring the target reference as well as adding authentication credentials.
+
+Once a target repository and reference is configured, Flipt will poll the source repository on a periodic cadence.
+This cadence is also configurable and defaults to 30 seconds.
+
+Flipt will follow the configured reference (e.g. branch name) and keep up to date with new changes.
+
+<Tabs>
+  <Tab title="Environment Variable">
+
+    ```bash
+    FLIPT_EXPERIMENTAL_FILESYSTEM_STORAGE_ENABLED=true
+    FLIPT_STORAGE_TYPE="git"
+    FLIPT_STORAGE_GIT_REPOSITORY="https://github.com/predictab.le/config.git"
+    FLIPT_STORAGE_GIT_REF="main"
+    FLIPT_STORAGE_GIT_POLL_INTERVAL="30s"
+    # for private repository access
+    FLIPT_STORAGE_GIT_AUTHENTICATION_BASIC_USERNAME=...
+    FLIPT_STORAGE_GIT_AUTHENTICATION_BASIC_PASSWORD=...
+    ```
+
+  </Tab>
+  <Tab title="Configuration Yaml">
+
+    ```yaml
+    experimental:
+      filesystem_storage:
+        enabled: true
+
+    storage:
+      type: git
+      git:
+        repository: "https://github.com/predictab.le/config.git"
+        ref: "main"
+        poll_interval: "30s"
+        authentication:
+          basic:
+            username: ...
+            password: ...
+          token:
+            access_token: ...
+    ```
+
+  </Tab>
+</Tabs>
+
+#### Authentication
+
+The `git` type backend supports both `basic` and `token` based authentication schemes.
+These sections are mutually exclusive. Flipt will fail to initialize with a validation error if you attempt to configure both.
```

Comment:
I looked at the code, and I am not sure if this is currently true, I apologize, I can add the validation logic for that: https://github.com/flipt-io/flipt/blob/main/internal/config/storage.go#L85-L98

Reply:
Thats no worries. I can adjust this to match current expectations. I don't think it super matters. I can just say that basic takes presedence.